### ISSUE TITLE
TypeScript typings of `EqualityFn`

### ISSUE THREAD

**Issue by SanderDeWaal1992** [alexreardon/memoize-one#73]
The arguments of `EqualityFn` are defined as being `unknown` even though they are known (i.e. the same as ...newArgs of ResultFn).

Our build did fail today due to the recently added TypeScript types. My suggestion would be to type the args of `EqualityFn` as any (just like the typings on DefinitelyTyped did which does make that users do not have to cast the unknown type) or to strong-type it, e.g.:

```
 export type EqualityFn<Args = any> = (
  newArgs: Args,
  lastArgs: Args,
) => boolean;

export default function memoizeOne<
  ResultFn extends (this: any, ...newArgs: T) => ReturnType<ResultFn>, 
  T extends any[]
>(resultFn: ResultFn, isEqual: EqualityFn<T> = areInputsEqual): ResultFn {
...
```

**Comment by alexreardon** [comment]
I'll get onto this soon

**Comment by alexreardon** [comment]
I suspect this would be sufficient:

```
type EqualityFn = (newArgs: ReadonlyArray<any>, lastArgs: ReadonlyArray<any>) => boolean;
```

Do you think the readonly part will cause issues?

**Comment by alexreardon** [comment]
having a strong type on the equality function might be a bit lame for consumers

**Comment by SanderDeWaal1992** [comment]
> 
> 
> I suspect this would be sufficient:
> 
> ```
> type EqualityFn = (newArgs: ReadonlyArray<any>, lastArgs: ReadonlyArray<any>) => boolean;
> ```
> 
> Do you think the readonly part will cause issues?

I made an example snippet:

```
const test = memoizeOne((
	arg1: string
) => {
	return "test" + arg1;
}, (
	a: Readonly<[string]>,
	b: Readonly<[string]>,
) => {
	return someCustomComparison(a as [string], b as [string]);
});
```

This will result in TS error:

> Argument of type '(a: readonly [string], b: readonly [string]) => any' is not assignable to parameter of type 'EqualityFn'.
>   Types of parameters 'a' and 'newArgs' are incompatible.
>     Property '0' is missing in type 'readonly any[]' but required in type 'readonly [string]'.ts(2345)

I'm not sure how I  -as a user- could type it correctly. I think its obvious for users that the arguments should not be adjusted and that just 'any'  should be fine.

**Comment by alexreardon** [comment]
I think this should be fine then:

```ts
type EqualityFn = (newArgs: any[], lastArgs: any[]) => boolean;
```

it is fairly loose, but i don't think the friction is worth typing it stricter for now

**Comment by alexreardon** [comment]
Fixed in 5.1.1
